src/Momento.Sdk/Exceptions/LimitExceededException.cs

```diff
@@ -1,15 +1,91 @@
 {
     /// <include file="../docs.xml" path='docs/class[@name="SdkException"]/constructor/*' />
-    public LimitExceededException(string message, MomentoErrorTransportDetails transportDetails, Exception? e = null) : base(MomentoErrorCode.LIMIT_EXCEEDED_ERROR, message, transportDetails, e)
+    public LimitExceededException(string message, MomentoErrorTransportDetails transportDetails, RpcException? e = null) : base(MomentoErrorCode.LIMIT_EXCEEDED_ERROR, message, transportDetails, e)
     {
-        this.MessageWrapper = "Request rate exceeded the limits for this account.  To resolve this error, reduce your request rate, or contact us at support@momentohq.com to request a limit increase";
+        var errMetadata = e?.Trailers.Get("err")?.Value;
+        if (errMetadata != null) {
+            this.MessageWrapper = errMetadata switch
+            {
+                "topic_subscriptions_limit_exceeded" => LimitExceededMessageWrapper.TopicSubscriptionsLimitExceeded.Value,
+                "operations_rate_limit_exceeded" => LimitExceededMessageWrapper.OperationsRateLimitExceeded.Value,
+                "throughput_rate_limit_exceeded" => LimitExceededMessageWrapper.ThroughputRateLimitExceeded.Value,
+                "request_size_limit_exceeded" => LimitExceededMessageWrapper.RequestSizeLimitExceeded.Value,
+                "item_size_limit_exceeded" => LimitExceededMessageWrapper.ItemSizeLimitExceeded.Value,
+                "element_size_limit_exceeded" => LimitExceededMessageWrapper.ElementSizeLimitExceeded.Value,
+                _ => LimitExceededMessageWrapper.UnknownLimitExceeded.Value,
+            };
+        } else {
+            var lowerCasedMessage = message.ToLower();
+            this.MessageWrapper = LimitExceededMessageWrapper.UnknownLimitExceeded.Value;
+            if (lowerCasedMessage.Contains("subscribers")) {
+                this.MessageWrapper = LimitExceededMessageWrapper.TopicSubscriptionsLimitExceeded.Value;
+            } else if (lowerCasedMessage.Contains("operations")) {
+                this.MessageWrapper = LimitExceededMessageWrapper.OperationsRateLimitExceeded.Value;
+            } else if (lowerCasedMessage.Contains("throughput")) {
+                this.MessageWrapper = LimitExceededMessageWrapper.ThroughputRateLimitExceeded.Value;
+            } else if (lowerCasedMessage.Contains("request limit")) {
+                this.MessageWrapper = LimitExceededMessageWrapper.RequestSizeLimitExceeded.Value;
+            } else if (lowerCasedMessage.Contains("item size")) {
+                this.MessageWrapper = LimitExceededMessageWrapper.ItemSizeLimitExceeded.Value;
+            } else if (lowerCasedMessage.Contains("element size")) {
+                this.MessageWrapper = LimitExceededMessageWrapper.ElementSizeLimitExceeded.Value;
+            } else {
+                this.MessageWrapper = LimitExceededMessageWrapper.UnknownLimitExceeded.Value;
+            }
+        }
     }
 }
```

Comment:
For separation of concerns, we let the constructor initialize the instance and separate out resolving the `MessageWrapper`. We can make a static method `CreateWithMessageWrapper` (or name of your choice) that resolves the `MessageWrapper`, then returns an instance with that set (by having a constructor that takes the `MessageWrapper`.